lib/command.js

```diff
@@ -230,14 +230,19 @@ module.exports = function command (yargs, usage, validation, globalMiddleware) {
 
       innerArgv = applyMiddleware(innerArgv, yargs, middlewares, false)
 
-      const handlerResult = isPromise(innerArgv)
-        ? innerArgv.then(argv => commandHandler.handler(argv))
-        : commandHandler.handler(innerArgv)
+      let handlerResult
+      if (isPromise(innerArgv)) {
+        yargs.getUsageInstance().cacheHelpMessage()
```

Comment:
I do not understand why you cache help message here? Shouldn't caching it before `handlerResult.catch` below be enough? Removing this line does not break any test.